### PR TITLE
SC-9143 implemented filter function on teachers dashbard and integrat…

### PR DESCRIPTION
**Pull request by martynakrysinska**
# Short Description

This PR contains: 
- v-autocomplete for filtering homeworks (student and teacher)
- translations
- grouping of tests for homeworks store
- validator for vHomeworkItemTeacher

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

**Review by dominikbruechner — CHANGES_REQUESTED** [review]
Just some minor things we can also discuss - and no storybook entries yet 🙈 I will be creating all of those, so you can focus on the other issues and wrap your day up before the vacation 🌻

File: src/store/homeworks.js [hunk 1]
```diff
@@ -18,6 +24,14 @@ const module = {
 				commit("setLoading", false);
 			}
 		},
+		updateFilter: function ({ commit }, courses) {
+			commit("setFilter", courses);
+		},
+	},
```
[the remark this answers is not in the document]
Comment (martynakrysinska) [reply src/store/homeworks.js:30]: I was thinking the same, but I'm afraid that calling mutations directly from components might be a bad practice. Let's do some reading about it?

File: src/store/homeworks.js [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
 const base = serviceTemplate("homework");
 
-const module = {
+const baseState = base.state();
+
+const module = mergeDeep(base, {
+	state: () =>
+		mergeDeep(baseState, {
+			courseFilter: [],
+		}),
```
[the remark this answers is not in the document]
Comment (martynakrysinska) [reply src/store/homeworks.js:11]: I couldn't make it work with the new local state (courseFilter) anymore.

**Comment by dominikbruechner** [comment]
@schul-cloud-bot update screenshots

**Review by dominikbruechner — APPROVED** [review]

**Comment by dominikbruechner** [comment]
@schul-cloud-bot update screenshots

**Pull request closed without merging**